OB-1: Sum functionality added

davidburgos opened the pull request:
Support for sum two integer values.
Testing Done: Sony Xperia Z5
Signed-off-by: David Burgos david.burgos@globant.com

davidburgos committed:
OB-1: Sum functionality added
Support for sum two integer values.
Testing Done: Sony Xperia Z5
Signed-off-by: David Burgos david.burgos@globant.com

andresuarezz26 commented on app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java:
```diff
@@ -0,0 +1,105 @@
+
+    private void onSolveButtonPressed() {
+
+        int result = 0;
```
In the project's checkstyle if you put the 0, it is marked as a warning named "Magic numbers". Please create a constant (private static final int ZERO_INT = 0;) or you can create a class to have all the constants.

andresuarezz26 commented on app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java:
```diff
@@ -0,0 +1,105 @@
+                result = model.getFirstValue() + model.getSecondValue();
+                break;
+            case UNKNOWN:
+                result = -1;
```
Magic number

davidburgos committed:
OB-1: Add constants
Add constants for int value
Testing Done: Sony Xperia Z5
Signed-off-by: David Burgos david.burgos@globant.com

andresuarezz26 commented on app/src/main/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenter.java:
```diff
@@ -0,0 +1,105 @@
+import com.davidburgos.calculator.util.bus.observers.CalculatorOperatorBusObserver;
+import com.davidburgos.calculator.util.bus.observers.CalculatorSecondValueBusObserver;
+
+public class CalculatorPresenter {
```
You must do the unit test for the Presenter

davidburgos replied:
Done

davidburgos committed:
OB-1: Add unit test for Presenter Class.
Presenter class has been tested with JUnit4 and mockito.
Testing Done: Sony Xperia Z5
Signed-off-by: David Burgos david.burgos@globant.com

andresuarezz26 commented on app/src/main/java/com/davidburgos/calculator/MainActivity.java:
```diff
@@ -1,15 +1,32 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        this.presenter = new CalculatorPresenter(new CalculatorModel(), new CalculatorView(this));
```
The reserved word this is redundant

davidburgos replied:
Done

andresuarezz26 commented on app/src/main/java/com/davidburgos/calculator/mvp/model/CalculatorModel.java:
```diff
@@ -0,0 +1,32 @@
+public class CalculatorModel {
+
+    private int firstValue = 0;
+    private int secondValue = 0;
```
These are magic numbers for the checkstyle. Please add constants for ZERO_INT

davidburgos replied:
Done

andresuarezz26 commented on app/src/main/java/com/davidburgos/calculator/mvp/view/CalculatorView.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    public void setResult(String value) {
+        this.result.setText(value);
```
this is redundant. You can see when to use this in the following post: https://stackoverflow.com/questions/2411270/when-should-i-use-this-in-a-class

davidburgos replied:
Done

davidburgos committed:
OB-1: Get rid of redundant code
redundant code has been deleted and a new constant value was created
Testing Done: Sony Xperia Z5
Signed-off-by: David Burgos david.burgos@globant.com

davidburgos committed:
OB-1: Refactor for Bus event classes
New test case added and refactor for bus classes.
Testing Done: Sony Xperia Z5
Signed-off-by: David Burgos david.burgos@globant.com

Signed-off-by: david.burgos <david.burgos@globant.com>

andresuarezz26 commented on app/src/test/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenterTest.java:
```diff
@@ -0,0 +1,55 @@
+    @Test
+    public void shouldReturnSumOfTwoIntValues() throws Exception {
+        when(model.getOperator()).thenReturn(Operators.PLUS);
+        when(model.getFirstValue()).thenReturn(10);
```
Remember the magic numbers. Is a little annoying but is for not generate checkstyle warnings

andresuarezz26 commented on app/src/test/java/com/davidburgos/calculator/mvp/presenter/CalculatorPresenterTest.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Test
+    public void shouldReturnSumOfTwoIntValues() throws Exception {
```
The test is fine! good work

davidburgos committed:
OB-1: Add extra Operators
Add extra operators and create constants for CalculatorPresenterTest
Testing Done: Sony Xperia Z5
Signed-off-by: David Burgos david.burgos@globant.com

Signed-off-by: david.burgos <david.burgos@globant.com>
